Modules/_pickle.c

```diff
@@ -518,9 +518,8 @@ Pdata_grow(Pdata *self)
 }
 
 static int
-Pdata_stack_underflow(Pdata *self)
+Pdata_stack_underflow(Pdata *self, PickleState *st)
```

Comment:
Would it be possible to pass the state as the first argument, as passing a context? Same remark for the whole PR.

Reply:
@vstinner sure, I didn't want to break the CPython convention of passing self first but if this is your preference I'll change it